The main goal was to see if this dataset held additional data that Transparent California did not include. I was told to look for things like Tenure Status and Positions of Faculty

In [ ]:
import polars as pl
import duckdb

conn = duckdb.connect()

# Read combined parquet file
gcc_df = pl.read_parquet("../data/GCC.parquet")

transparent_df = pl.read_parquet("../data/faculty-salaries.parquet")



In [34]:
# We really only care about faculty data
df = conn.sql(""" 
        SELECT * FROM gcc_df WHERE Position ILIKE '%professor%'
                            OR Position ILIKE '%lecturer%'
                            OR Position ILIKE '%faculty%'
                            OR Position ILIKE '%instructor%'
        """).pl()

df.sample(25)


Year,EmployerType,EmployerName,DepartmentOrSubdivision,Position,ElectedOfficial,Judicial,OtherPositions,MinPositionSalary,MaxPositionSalary,ReportedBaseWage,RegularPay,OvertimePay,LumpSumPay,OtherPay,TotalWages,DefinedBenefitPlanContribution,EmployeesRetirementCostCovered,DeferredCompensationPlan,HealthDentalVision,TotalRetirementAndHealthContribution,PensionFormula,EmployerURL,EmployerPopulation,LastUpdatedDate,EmployerCounty,SpecialDistrictActivities,IncludesUnfundedLiability,SpecialDistrictType
i64,str,str,str,str,bool,bool,str,f64,f64,str,f64,f64,f64,f64,f64,f64,i64,i64,f64,f64,str,str,str,date,str,str,bool,str
2024,"""California State University""","""State University, Sacramento""",null,"""Instructional Faculty - 12 Mon…",false,false,"""Ft Faculty Unit - Additional E…",85524.0,182532.0,null,127465.0,0.0,0.0,0.0,127465.0,34941.71,0,0,83.52,35025.23,"""Public Employees Retirement Sy…","""https://www.csus.edu/administr…",null,2025-07-29,"""Sacramento""",null,true,null
2024,"""Community College District""","""Los Angeles Community College …","""ELAC Child, Family, and Educat…","""Instructor, Adjunct""",false,false,null,42868.0,52824.0,null,23842.0,0.0,0.0,2426.0,26268.0,4922.0,0,0,0.0,4922.0,"""2% @ 60""","""https://www.laccd.edu/Employme…",null,2025-09-30,"""Los Angeles""",null,false,null
2024,"""University of California""","""University of California, San …","""University of California - San…","""K-12 Instructor-Ay""",false,false,null,0.0,0.0,null,13551.0,0.0,0.0,0.0,13551.0,619.0,0,0,2277.0,2896.0,"""2016 TIER""","""https://ucnet.universityofcali…",null,2025-09-30,"""San Diego""",null,false,null
2024,"""Community College District""","""Santa Monica Community College…","""History""","""Faculty Tech-Academic""",false,false,null,83118.0,176414.0,null,182735.0,0.0,0.0,4000.0,186735.0,35614.0,0,0,21505.0,57119.0,"""2% @ 60""","""https://www.smc.edu/administra…",null,2025-09-30,"""Los Angeles""",null,false,null
2024,"""California State University""","""State University, Chico""",null,"""Lecturer - Academic Year, Rang…",false,false,"""Special Consultant, Range 0""",66084.0,80124.0,null,46776.72,0.0,0.0,11799.6,58576.32,16649.79,0,0,1763.52,18413.31,"""Public Employees Retirement Sy…","""http://www.csuchico.edu/hr/""",null,2025-07-29,"""Butte""",null,true,null
…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…,…
2024,"""California State University""","""State University, San Francisc…",null,"""Lecturer - Academic Year, Rang…",false,false,null,74652.0,158688.0,null,21807.61,0.0,0.0,0.0,21807.61,6445.18,0,0,0.0,6445.18,"""Public Employees Retirement Sy…","""https://hr.sfsu.edu/""",null,2025-07-29,"""San Francisco""",null,true,null
2024,"""California State University""","""State University, San Bernardi…",null,"""Lecturer - Academic Year, Rang…",false,false,null,66084.0,80124.0,null,7893.55,0.0,0.0,0.0,7893.55,0.0,0,0,20.88,20.88,"""Part-Time Seasonal And Tempora…","""https://www.csusb.edu/human-re…",null,2025-07-29,"""San Bernardino""",null,true,null
2024,"""California State University""","""State University, Chico""",null,"""Instructional Faculty - Academ…",false,false,null,81900.0,174276.0,null,75604.0,0.0,0.0,9614.0,85218.0,26265.92,0,0,8873.73,35139.65,"""Public Employees Retirement Sy…","""http://www.csuchico.edu/hr/""",null,2025-07-29,"""Butte""",null,true,null


In [35]:
# From the random sampling, it seems a lot of DeptOrSubdivision is missing, lets see how much stuff is missing in general
for col in df.columns:
    print(f"Percent of data missing from {col}: {((df[col].null_count() / df.height) * 100):.2f}%")

Percent of data missing from Year: 0.00%
Percent of data missing from EmployerType: 0.00%
Percent of data missing from EmployerName: 0.00%
Percent of data missing from DepartmentOrSubdivision: 44.86%
Percent of data missing from Position: 0.00%
Percent of data missing from ElectedOfficial: 0.00%
Percent of data missing from Judicial: 0.00%
Percent of data missing from OtherPositions: 66.63%
Percent of data missing from MinPositionSalary: 1.86%
Percent of data missing from MaxPositionSalary: 1.86%
Percent of data missing from ReportedBaseWage: 100.00%
Percent of data missing from RegularPay: 0.00%
Percent of data missing from OvertimePay: 0.81%
Percent of data missing from LumpSumPay: 0.55%
Percent of data missing from OtherPay: 0.08%
Percent of data missing from TotalWages: 0.00%
Percent of data missing from DefinedBenefitPlanContribution: 0.07%
Percent of data missing from EmployeesRetirementCostCovered: 2.15%
Percent of data missing from DeferredCompensationPlan: 0.11%
Percent of dat

Seems like the most signifcant ones that we would care about are:  
    Percent of data missing from DepartmentOrSubdivision: 44.86%
    Percent of data missing from OtherPositions: 66.63%



Does not seem to fill in the some of the gaps missing in Transparent California very well, so I am not that optimisitc about using this data set in general.

That being said, we might be able to estimate what position they hold based on how much their salary is. This might be a big stretch though.

EX from dataset for CSU Fullerton Lecturer:

"Lecturer - Academic Year, Range 2"
Min Salary: 66084.0
Max Salary:  80124.0

"Lecturer - Academic Year, Range 3"
Min Salary: 74652.0
Max Salary: 158688.0

If we are trying to find out a CSU Fullerton Professor's position, and their salary is above 80124.0, then they are range 3

Here is what Claude said ranges mean:

- Range 1: Lecturer A (entry-level, e.g. no terminal degree required)
- Range 2: Lecturer B (roughly comparable experience level to Assistant Professor)
- Range 3: Lecturer C (roughly comparable to Associate Professor)
- Range 4: Lecturer D (most senior/experienced, comparable to Full Professor)

I am going to see if maybe we can join Transparent California and this GCC dataset, using TotalWages and totalpaybenefits. I would assume that this is how TransparentCalifornia does it  

In [36]:
combined_df = conn.sql("""
    WITH gcc_agg AS (
        SELECT EmployerName, Position, TotalWages, Year
        FROM gcc_df
    ),
    transparent_agg AS (
        SELECT department, job, totalpay, year
        FROM transparent_df
        WHERE year = 2024
          AND department IS NOT NULL
    )
    SELECT
        g.EmployerName,
        g.Position,
        g.TotalWages,
        g.Year,
        t.department,
        t.job,
        t.totalpay
    FROM transparent_agg t
    JOIN gcc_agg g
        ON g.totalWages = t.totalpay
""").pl()

print(f"Matched {combined_df.height} rows")
combined_df.head(25)


Matched 29 rows


EmployerName,Position,TotalWages,Year,department,job,totalpay
str,str,f64,i64,str,str,f64
"""Desert Community College Distr…","""Instructor""",208634.0,2024,"""Applied Science and Business""","""Instructor""",208634.0
"""Desert Community College Distr…","""Instructor""",236240.0,2024,"""Applied Science and Business""","""Instructor""",236240.0
"""Desert Community College Distr…","""Instructor""",234391.0,2024,"""Applied Science and Business""","""Instructor""",234391.0
"""Desert Community College Distr…","""Instructor""",206084.0,2024,"""Communication and Humanities""","""Instructor""",206084.0
"""Desert Community College Distr…","""Instructor""",224069.0,2024,"""Communication and Humanities""","""Instructor""",224069.0
…,…,…,…,…,…,…
"""Desert Community College Distr…","""Instructor""",198799.0,2024,"""Social Science and Arts""","""Instructor""",198799.0
"""Desert Community College Distr…","""Instructor""",212183.0,2024,"""Social Science and Arts""","""Instructor""",212183.0
"""Desert Community College Distr…","""Interim Dean, Student Support …",229649.0,2024,"""Vice President - Student Servi…","""Interim Dean, Student Support …",229649.0


I will want someone to help verify if this is a successful join